pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -120,3 +122,23 @@ func (r *TokenREST) GroupVersionKind(containingGV schema.GroupVersion) schema.Gr
 type getter interface {
 	Get(ctx genericapirequest.Context, name string, options *metav1.GetOptions) (runtime.Object, error)
 }
+
+// newContext return a copy of ctx in which new RequestInfo is set.
+func newContext(ctx genericapirequest.Context, resource, name string, gvk schema.GroupVersionKind) genericapirequest.Context {
+	oldInfo, _ := genericapirequest.RequestInfoFrom(ctx)
+	newInfo := newRequestInfo(oldInfo, resource, name, gvk)
+	return genericapirequest.WithRequestInfo(ctx, &newInfo)
+}
+
+func newRequestInfo(old *genericapirequest.RequestInfo, resource, name string, gvk schema.GroupVersionKind) genericapirequest.RequestInfo {
```

Comment:
it would be easier to read if this function was inlined.

Reply:
Done.